generators/new_collateral/initializer/generator.go

```diff
@@ -0,0 +1,70 @@
+package initializer
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/dave/jennifer/jen"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/types"
+)
+
+type IGenerate interface {
+	GenerateFlipInitializer() error
+	GenerateMedianInitializer() error
+}
+
+type Generator struct {
+	Collateral                types.Collateral
+	MedianInitializerRequired bool
+}
+
+var initializerFileName = "initializer.go"
+
+func (g *Generator) GenerateFlipInitializer() error {
+	initializer := g.createInitializer(g.Collateral.GetFlipContractName(), "flip")
+	//create the path to the initializer file
+	path := g.Collateral.GetAbsoluteFlipStorageInitializersDirectoryPath()
+	mkDirErr := os.MkdirAll(path, os.ModePerm)
```

Comment:
could go either way here but I wonder if we might want to consider extracting/wrapping these filesystem IO operations and mocking them in the these tests? Not sure if that gets super complex when calling `Save` on the object returned by `jen.NewFile` 🤔

Reply:
I really like this idea, but may hold off on it for a bit - now that I've added a test for the contents of the initializer, I kind of like being able to see the output of calling `Save` on the `jen.NewFile`. I also think that having a `jen` wrapper may be cool but was struggling to find a good way to do that without needing to mock out each of the render methods, eg `Var()`, `Id()`, `Qual()`, etc.